fdbserver/RestoreApplier.actor.cpp

```diff
@@ -0,0 +1,263 @@
+			self->kvOps[commitVersion].push_back_deep(self->kvOps[commitVersion].arena(), mutation);
+			numMutations++;
+		}
+
+		// Notify the same actor and unblock the request at the next version
+		if (req.isRangeFile) {
+			self->rangeVersion.set(req.version);
+		} else {
+			self->logVersion.set(req.version);
+		}
+	}
+
+	req.reply.send(RestoreCommonReply(self->id()));
+	return Void();
+}
+
+ACTOR Future<Void> applyToDB(Reference<RestoreApplierData> self, Database cx) {
+	state std::string typeStr = "";
+
+	// Assume the process will not crash when it apply mutations to DB. The reply message can be lost though
+	if (self->kvOps.empty()) {
+		TraceEvent("FastRestore").detail("ApplierApplyToDBEmpty", self->id());
+		return Void();
+	}
+	std::map<Version, Standalone<VectorRef<MutationRef>>>::iterator begin = self->kvOps.begin();
+	std::map<Version, Standalone<VectorRef<MutationRef>>>::iterator end = self->kvOps.end();
+	end--;
+	ASSERT_WE_THINK(end != self->kvOps.end());
+	TraceEvent("FastRestore")
+	    .detail("ApplierApplyToDB", self->id())
+	    .detail("FromVersion", begin->first)
+	    .detail("EndVersion", end->first);
+
+	self->sanityCheckMutationOps();
+
+	state std::map<Version, Standalone<VectorRef<MutationRef>>>::iterator it = self->kvOps.begin();
+	state std::map<Version, Standalone<VectorRef<MutationRef>>>::iterator prevIt = it;
+	state int index = 0;
+	state int prevIndex = index;
+	state int count = 0;
+	state Reference<ReadYourWritesTransaction> tr(new ReadYourWritesTransaction(cx));
+	state int numVersion = 0;
+	state double transactionSize = 0;
+	loop {
+		try {
+			tr->reset();
+			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			tr->setOption(FDBTransactionOptions::LOCK_AWARE);
+			transactionSize = 0;
+
+			for (; it != self->kvOps.end(); ++it) {
+				numVersion++;
+				//TraceEvent("FastRestore").detail("Applier", self->id()).detail("ApplyKVsToDBVersion", it->first);
+				state MutationRef m;
+				for (; index < it->second.size(); ++index) {
+					m = it->second[index];
+					if (m.type >= MutationRef::Type::SetValue && m.type <= MutationRef::Type::MAX_ATOMIC_OP)
+						typeStr = typeString[m.type];
+					else {
+						TraceEvent(SevError, "FastRestore").detail("InvalidMutationType", m.type);
+					}
+
+					if (m.type == MutationRef::SetValue) {
+						tr->set(m.param1, m.param2);
+					} else if (m.type == MutationRef::ClearRange) {
+						KeyRangeRef mutationRange(m.param1, m.param2);
+						tr->clear(mutationRange);
+					} else if (isAtomicOp((MutationRef::Type)m.type)) {
+						tr->atomicOp(m.param1, m.param2, m.type);
+					} else {
+						TraceEvent(SevError, "FastRestore")
+						    .detail("UnhandledMutationType", m.type)
+						    .detail("TypeName", typeStr);
+					}
+					++count;
+					transactionSize += m.expectedSize();
+
+					if (transactionSize >= opConfig.transactionBatchSizeThreshold) { // commit per 1000 mutations
+						wait(tr->commit());
+						tr->reset();
```

Comment:
Why the reset?

Reply:
The transaction handler `tr` did the commit above. This is the next new transaction, so I called the `reset()` to reset the state of the transaction.